httpclient.go

```diff
@@ -187,6 +190,14 @@ func WithContentType(ct string) Opt {
 	}
 }
 
+// WithHeader is a client option for setting custom http header(s) for each request
```

Comment:
@ghouscht Please add comment that Content-Type and Accept headers will always be overwritten